modin/pandas/groupby.py

```diff
@@ -659,16 +659,20 @@ def apply(self, func, *args, **kwargs):
         if not isinstance(func, BuiltinFunctionType):
             func = wrap_udf_function(func)
 
-        return self._check_index(
-            self._wrap_aggregation(
-                qc_method=type(self._query_compiler).groupby_agg,
-                numeric_only=False,
-                agg_func=func,
-                agg_args=args,
-                agg_kwargs=kwargs,
-                how="group_wise",
-            )
+        apply_res = self._wrap_aggregation(
+            qc_method=type(self._query_compiler).groupby_agg,
+            numeric_only=False,
+            agg_func=func,
+            agg_args=args,
+            agg_kwargs=kwargs,
+            how="group_wise",
         )
+        reduced_index = pandas.Index([MODIN_UNNAMED_SERIES_LABEL])
+        if not isinstance(apply_res, Series) and apply_res.columns.equals(
+            reduced_index
+        ):
+            apply_res = apply_res.squeeze()
```

Comment:
```suggestion
            apply_res = apply_res.squeeze(axis=1)
```
Otherwise, it can accidentally squeeze the result into a scalar:
```python
>>> df = pd.DataFrame({"a": [1]})
>>> df
   a
0  1
>>> df.squeeze()
1
>>> df.squeeze(axis=1)
0    1
Name: a, dtype: int64
```